packages/rate-limit-controller/src/RateLimitController.ts

```diff
@@ -122,11 +142,11 @@ export class RateLimitController<
     this.rateLimitCount = rateLimitCount;
 
     this.messagingSystem.registerActionHandler(
-      `${name}:call` as const,
+      `${name}:call`,
       (
         origin: string,
         type: keyof RateLimitedApis,
-        ...args: Parameters<RateLimitedApis[keyof RateLimitedApis]['method']>
+        ...args: Parameters<RateLimitedApis[typeof type]['method']>
```

Comment:
`args` should represent the result of querying `RateLimitedApis` with the specific `type` passed in as the second argument, not all possible Api types stored in state. 

`keyof RateLimitedApis` is only the constraint for the `type` param, not its type.